refactor mission components

SalmaAnany opened the pull request.

SalmaAnany committed:
Remove TimeProgress and refactor mission components

TimeProgress component is removed, and its logic is integrated into MissionChoresList to simplify the mission's time and progress tracking. Components such as TotalPointsEarned, RedeemRewardButton, FinishMissionButton, and Sidebar are updated for improved styling and functionality. A new dynamic logo is added, enhancing the interface.

Msambere commented on public/logo.svg:
Can we add the logo to the login page too?

Msambere commented on src/Components/MissionPage/ActiveMission/ActiveMission.tsx:
```diff
@@ -153,34 +154,60 @@ const ActiveMission = ({
-
-        {/* Middle Panel - Time & Progress */}
-        <Grid size={4}>
-          <TimeProgress
-            chores={chores}
-            missionFinished={missionFinished}
-            onTimeRunOut={handleFinishMission}
-          />
```
Is there a reason this was deleted? Because it does seem like the timer is no longer working.

Msambere reviewed (commented):
Can we add the logo to the login page as well?

momofAnAl committed:
fixing the set useState in my totalUnredeemPoints

momofAnAl committed:
change the time elapsed to a string set up

momofAnAl committed:
change the usestate to App and use it global when start the Mission

momofAnAl committed:
update Total Point Logic

SalmaAnany committed:
Refactor mission flow to enable dynamic mission management

Updated components to facilitate mission creation, tracking, and state sharing between views. Introduced new props and state handling for managing mission data across `UserProfileView`, `RecommendMissionCard`, and `MissionView`. Improved mission progression logic and cleaned up redundant variables for streamlined behavior and better maintainability.
